src/main.rs

```diff
@@ -753,7 +755,7 @@ pub(crate) fn single_desugar_test(source: &str) -> Result<String, VerifyError> {
     let mut new_source_units: Vec<Item<SourceUnit>> = vec![];
     source_unit
         .enter()
-        .apply_encodings(&mut tcx, &mut new_source_units)?;
+        .apply_encodings(&mut tcx, &mut new_source_units, &HashMap::new())?;
```

Comment:
This additional HashMap parameter seems rather magical. Why not hide it in the `apply_encodings` call or alternatively create a `LoopChecker` which contains this state?